### PR TITLE
Lock Update NFT in UpdateContract Box

### PR DESCRIPTION
**Pull request by SethDusek**
For #50

### CODE REVIEW

**Comment by SethDusek** [comment]
Not sure what's up with the code coverage test. Did I mess something up while merging?

**Comment by greenhat** [comment]
> Not sure what's up with the code coverage test. Did I mess something up while merging?

Nope. coveralls.io was down. I restarted the job.

**Comment by SethDusek** [comment]
Thanks! Anyways, this PR should be good to go now. I've also modified the tests a bit so we can inspect the bootstrap transactions in unit tests and make sure everything is as it should be. Right now I'm just checking for the right parameters in the update contract though

**Comment by greenhat** [comment]
Could you please rebase? Probably one of the recently merged PRs caused the conflict.

**Comment by SethDusek** [comment]
Done! The new tests actually caught a mistake I made in the commit, which I think is pretty neat

File: core/src/contracts/update.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@ use thiserror::Error;
 
 #[derive(Clone)]
 pub struct UpdateContract {
-    ergo_tree: ErgoTree,
+    pub ergo_tree: ErgoTree,
```
Comment (greenhat) [inline core/src/contracts/update.rs:12]: Could you please add a getter for `ergo_tree` property like in other contracts? I'm still not sure if we'll keep `ErgoTree` or generate it on request, so it's better to keep it private.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking great! I appreciate the test! Just one minor request.

**Review by greenhat — APPROVED** [review]
Great! Thank you!

**Pull request merged**